src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -180,7 +181,16 @@ void GetLocationInfo(out FileLinePositionSpan originalLineInfo, out FileLinePosi
             }
             else
             {
-                originalLineInfo = location.GetLineSpan();
+                try
+                {
+                    originalLineInfo = location.GetLineSpan();
+                }
+                catch (Exception e) when (FatalError.ReportWithDumpAndCatch(e, ErrorSeverity.Critical))
```

Comment:
is tehre no REportWithDumpAndPropagate?

Reply:
guess there isn't!

Reply:
i would explicitly add a comment stating this is for https://github.com/dotnet/roslyn/pull/74728